examples/typescript/aws-s3-static-website/main.ts

```diff
@@ -0,0 +1,98 @@
+import { Construct } from "constructs";
+import { App, TerraformStack, TerraformOutput, TerraformVariable } from "cdktf";
+import { AwsProvider, s3 } from '@cdktf/provider-aws';
+
+
+class MyStack extends TerraformStack {
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+
+    // Template Variables
+    const awsRegion = new TerraformVariable(this, 'awsRegion', {
+      type: 'string',
+      default: 'us-east-1',
+      description: 'The AWS Region to be use'
+    });
+
+    const bucketName = new TerraformVariable(this, 'bucketName', {
+      type: 'string',
+      default: 'test-bucket',
+      description: 'The name-prefix of the S3 bucket to be use'
+    });
+
+    // Provider Configuration
+    new AwsProvider(this, "aws", {
+      region: awsRegion.value,
+    });
+
+    // Create S3 Bucket
+    const s3bucket = new s3.S3Bucket(this, 'bucket', {
+      bucketPrefix: bucketName.value+'-',
+      tags: {
+        Name: bucketName.value,
+      },
+      forceDestroy: true
+    });
+
+    // Create S3 Bucket Policy to allow public access
+    new s3.S3BucketAcl(this, 'bucketAcl', {
+      bucket: s3bucket.id,
+      acl: 'public-read'
+    });
+
+    // Upload files to S3 Website
+    new s3.S3Object(this, 'indexObject', {
+      bucket: s3bucket.id,
+      contentType: "text/html",
+      key: 'index.html',
+      source: './website/index.html'
```

Comment:
I tried deploying the example to AWS and it failed with:
> Error opening S3 object source (./website/index.html): open ./website/index.html: no such file or directory

A TerraformAsset ([example](https://www.terraform.io/cdktf/concepts/assets#usage-example)) is required to ensure the file is available in the `cdktf.out` directory from where Terraform is applied.